enhancements/microshift/lvms-topolvm-downstream-replacement.md

```diff
@@ -0,0 +1,283 @@
+   * When External Configuration is not used, then not specifying any deviceClasses
+     will result in rejection of the LVMCluster specification.
+   * Note that while it is eventually planned to decouple the CSI stack from MicroShifts
+     core resources, this enhancement only targets the migration of the existing solution,
+     not the adoption of LVMS as an external component out-of-tree.
+8. MicroShift continues to run, and LVMS starts the vg-manager DaemonSet with all CSI bindings included.
+9. All existing configurations continue to work, with the following changes to log destinations
+   * Errors of TopoLVM in the node will now be written to the vg-manager Pod for all node-relevant logs and 
+     to the lvms-controller Pod for all CSI controller logs.
+   * Errors of LVMCluster reconciliation will be written to the lvms-controller Pod and Events
+     that are keyed to the LVMCluster resource.
+
+### API Extensions
+
+MicroShift will now introduce the `LVMCluster`, `LVMVolumeGroup` and `LVMVolumeGroupNodeStatus` CustomResources.
+All existing documentation of LVMS on the CR usages applies and can be cross-referenced if users
+want a simplified setup of their storage devices.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+N/A
+
+#### Standalone Clusters
+
+N/A
+
+#### Single-node Deployments or MicroShift
+
+This enhancement only applies to MicroShift.
+
+### Implementation Details/Notes/Constraints
+
+#### Enablement of configuration of no deviceClasses in LVMS
+
+Currently, LVMS does not allow for the configuration of no deviceClasses in the LVMCluster.
+This is a constraint that we will have to work around by allowing the user to specify
+an empty deviceClass in the LVMCluster and then coordinating that with the vg-manager.
+We have a prototype implementation that can be pushed into GA in time for the 4.17 release based on
+[OCPEDGE-830](https://issues.redhat.com//browse/OCPEDGE-830) that proves this is possible.
+
+##### Why is this a constraint? Why should we not introduce a migration Path for unmanaged deviceClasses to LVMCluster?
+
+The main issue with the current discrepancy between TopoLVM and LVMS is that TopoLVM allows for the configuration of deviceClasses with arbitrary options for supporting
+custom scenarios such as LVM RAID. This is not supported in LVMS (due to incompatibility with thin provisioning).
+
+If we would switch the stance to migrate existing MicroShift Installations fully to LVMCluster, we would need to
+maintain this migration while ensuring that existing RAID configurations are still supported. This is not feasible
+because a custom created volume group without constraints is introducing a lot of complexity to a potential test matrix
+that would explode the scope of LVMCluster testing and supported scenarios for OpenShift, which we expressively not want.
```

Comment:
What's happening in reality? Do we have any MicroShift customers who are utilizing the raid settings in the lvmd.yaml?